compiler/compile/OSRData.cpp

```diff
@@ -513,6 +514,388 @@ uint32_t TR_OSRCompilationData::getOSRStackFrameSize(uint32_t methodIndex)
+   {
+   TR_ASSERT(block->getSuccessors().size() == 1, "OSRCodeBlock should have one successor but block_%d has %d", block->getNumber(), block->getSuccessors().size());
+   if (comp()->getOption(TR_TraceOSR))
+      traceMsg(comp(), "buildDefiningMapForOSRCodeBlockAndPrepareForOSRCall block_%d\n", block->getNumber()); 
+   buildDefiningMap(block, osrCodeBlockMap, prepareForOSRCallMap);
+   if (block->getEntry()->getNode()->getByteCodeInfo().getCallerIndex() == -1)
+      return;
+   TR::Block *nextBlock = toBlock(block->getSuccessors().front()->getTo()); 
+   // Keep processing all blocks one after another until reaching the next OSRCodeBlock
+   if (!nextBlock->isOSRCodeBlock())
+      buildDefiningMapForBlock(nextBlock, osrCodeBlockMap);
+   }
+
+/*
+ * \brief This function builds the defining map for the given block and all the blocks after
+ * until reaching the `next OSRCodeBlock`. 
+ *
+ * If called from `buildDefiningMapForOSRCodeBlockAndPrepareForOSRCall`, the map is for
+ * an OSRCodeBlock and `next OSRCodeBlock` belongs to the next caller in the call chain of this 
+ * method. Note that it's possible that the immediate next caller doesn't have an OSRCodeBlock
+ * and the `next OSRCodeBlock` would belong to the caller's caller.
+ *
+ * If called from `buildDefiningMap`, the map is for the OSRCatchBlock, and the `next OSRCodeBlock`
+ * belongs to the same method as the OSRCatchBlock.
+ */
+void
+TR_OSRMethodData::buildDefiningMapForBlock(TR::Block *block, DefiningMap *blockMap)
+   {
+   TR_ASSERT(block->getSuccessors().size() == 1, "OSRCatchBlock should have one successor but block_%d has %d", block->getNumber(), block->getSuccessors().size());
+   if (comp()->getOption(TR_TraceOSR))
+      traceMsg(comp(), "buildDefiningMapForBlock block_%d\n", block->getNumber()); 
+   buildDefiningMap(block, blockMap);
+   TR::Block *nextBlock = toBlock(block->getSuccessors().front()->getTo()); 
+   if (!nextBlock->isOSRCodeBlock())
+      buildDefiningMapForBlock(nextBlock, blockMap);
+   }
+
+/*
+ * \brief This is the most basic function iterating through all store nodes and add entries to the DefiningMap to
+ * map each symbol to the group of symbols defining it
+ */
+void
+TR_OSRMethodData::buildDefiningMap(TR::Block *block, DefiningMap *blockMap, DefiningMap *prepareForOSRCallMap)
+   {
+   for (TR::TreeTop *tt = block->getEntry(); tt != block->getExit(); tt = tt->getNextRealTreeTop())
+      {
+      TR::SymbolReference * symRef = NULL;
+      TR::Node *node = tt->getNode();
+
+      if (comp()->getOption(TR_TraceOSR))
+         traceMsg(comp(), "buildDefiningMap node n%dn\n", node->getGlobalIndex());
+
+      if (node->getOpCode().isStoreDirect())
+         symRef = node->getSymbolReference();
+      else if (node->getOpCode().isStoreReg())
+         symRef = node->getRegLoadStoreSymbolReference();
+
+      if (symRef)
+         {
+         if (symRef->getSymbol()->isAutoOrParm())
+            {
+            TR_BitVector subTreeSymRefs;
```

Comment:
A `TR_BitVector` without a specified region will perform allocation in persistent memory. Should this be in the current stack region?

Reply:
Good catch - yes I think @ncough is right

Reply:
modified to use current stack region. thanks! @ncough